src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/fsm/SwitchSyncFsm.java

```diff
@@ -280,11 +256,20 @@ protected void computeMissingRules(SwitchSyncState from, SwitchSyncState to,
 
     protected void computeMisconfiguredRules(SwitchSyncState from, SwitchSyncState to,
                                              SwitchSyncEvent event, Object context) {
-        List<Long> reinstallRules = getReinstallDefaultRules();
+        Set<Long> reinstallRules = validationResult.getValidateRulesResult().getMisconfiguredRules();
         if (!reinstallRules.isEmpty()) {
-            log.info("Compute reinstall default rules (switch={}, key={})", switchId, key);
+            log.info("Compute reinstall rules (switch={}, key={})", switchId, key);
             try {
-                misconfiguredRules = commandBuilder.buildCommandsToReinstallRules(switchId, reinstallRules);
+                List<FlowSpeakerData> misconfiguredRulesToRemove = reinstallRules.stream()
+                        .map(this::findActualFlowByCookie)
+                        .collect(Collectors.toList());
+                toRemove.addAll(misconfiguredRulesToRemove);
+                List<FlowSpeakerData> misconfiguredRulesToInstall = validationResult.getExpectedEntries().stream()
+                        .filter(entry -> entry instanceof FlowSpeakerData)
+                        .map(entry -> (FlowSpeakerData) entry)
+                        .filter(flowEntry -> installedRulesCookies.contains(flowEntry.getCookie().getValue()))
+                        .collect(Collectors.toList());
+                toInstall.addAll(misconfiguredRulesToInstall);
```

Comment:
Why did you replace rule modification with delete and install?

There is a test case when it will fail:
1. modify some rule 123 on switch
2. validate switch - you will see one misconfigured flow
3. sync switch
`installedRulesCookies` will be empty because there is no missing rules.
`toRemove` will contain rule 123
`toInstall` will be empty because `installedRulesCookies` is empty and filter
`.filter(flowEntry -> installedRulesCookies.contains(flowEntry.getCookie().getValue()))`
will return nothing
4. after syncing you will have 1 missing rule. (instead of 1 misconfigured)

Reply:
It was already implemented as delete-install in previous version. This two step action was covered under the hood on FL side. It is used because we can't guarantee OpenFlow flow modification is usable for all misconfigured rule cases. E.g. wrong match on rule.

Test case fixed, docs updated.